File: cpp_src/simple_pub_sub/src/pub.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char **argv) {
    ros::init(argc, argv, "sim_pub");
    ros::NodeHandle nh;

    ros::Publisher std_pub = nh.advertise<std_msgs::String>("chatter", 1000);
    ros::Rate loop_rate(10);

    int count = 0;
    while (ros::ok()) {
        std_msgs::String msg;
        msg.data = "hello ycao " + std::to_string(count++);
        ROS_INFO("%s", msg.data.c_str());

        std_pub.publish(msg);

        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: pyhon_src/simple_pub_sub/scripts/pub.py
#! /usr/bin/env python3

import rospy
from std_msgs.msg import String


def main():
    rospy.init_node("sim_pub")
    pub = rospy.Publisher("chatter", String, queue_size=10)

    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        msg = String()
        msg.data = "hello ycao %s" %rospy.get_time()

        pub.publish(msg)

        rospy.loginfo(msg.data)

        rate.sleep()

if __name__ == "__main__":
    main()